diff -up mysql-5.5.28/mariadb.service.scl-env-check mysql-5.5.28/mariadb.service
--- mysql-5.5.31/mariadb.service.scl-env-check  2012-11-07 13:29:33.003742601 +0100
+++ mysql-5.5.31/mariadb.service        2012-11-07 14:06:23.090963675 +0100
@@ -27,11 +27,16 @@ Type=simple
 User=mysql
 Group=mysql
 
-ExecStartPre=/usr/libexec/mysqld-prepare-db-dir %n
+# Load collections set to enabled for this service
+EnvironmentFile=__SCL_SCRIPTS__/service-environment
+
+# We want to start server only inside "scl enable" invocation
+ExecStartPre=/usr/bin/scl-service $MARIADB55_SCLS_ENABLED /usr/bin/scl_enabled __SCL_NAME__
+ExecStartPre=/usr/bin/scl-service $MARIADB55_SCLS_ENABLED /usr/libexec/mysqld-prepare-db-dir %n
 # Note: we set --basedir to prevent probes that might trigger SELinux alarms,
 # per bug #547485
-ExecStart=/usr/bin/mysqld_safe --basedir=/usr
-ExecStartPost=/usr/libexec/mysqld-wait-ready $MAINPID
+ExecStart=/usr/bin/scl-service $MARIADB55_SCLS_ENABLED /usr/bin/mysqld_safe --basedir=/usr
+ExecStartPost=/usr/bin/scl-service $MARIADB55_SCLS_ENABLED /usr/libexec/mysqld-wait-ready $MAINPID
 
 # Give a reasonable amount of time for the server to start up/shut down
 TimeoutSec=300
diff -up mysql-5.5.31/scripts/mysqld_safe.sh.scl-env-check mysql-5.5.31/scripts/mysqld_safe.sh
--- mysql-5.5.31/scripts/mysqld_safe.sh.scl-env-check	2013-03-25 14:14:58.000000000 +0100
+++ mysql-5.5.31/scripts/mysqld_safe.sh	2013-05-02 14:46:21.577433667 +0200
@@ -11,6 +11,12 @@
 # mysql.server works by first doing a cd to the base directory and from there
 # executing mysqld_safe
 
+# we want start daemon only inside "scl enable" invocation
+if ! scl_enabled __SCL_NAME__ ; then
+    echo "Use \"scl enable __SCL_NAME__ 'service ...'\" invocation"
+    exit 1
+fi
+
 # Initialize script globals
 KILL_MYSQLD=1;
 MYSQLD=
diff -up mysql-5.5.31/scripts/mysql_install_db.sh.scl-env-check mysql-5.5.31/scripts/mysql_install_db.sh
--- mysql-5.5.31/scripts/mysql_install_db.sh.scl-env-check	2013-03-25 14:14:58.000000000 +0100
+++ mysql-5.5.31/scripts/mysql_install_db.sh	2013-05-02 14:46:21.585433669 +0200
@@ -18,6 +18,12 @@
 #
 # All unrecognized arguments to this script are passed to mysqld.
 
+# we want start daemon only inside "scl enable" invocation
+if ! scl_enabled __SCL_NAME__ ; then
+    echo "Use \"scl enable __SCL_NAME__ 'service ...'\" invocation"
+    exit 1
+fi
+
 basedir=""
 builddir=""
 ldata="@localstatedir@"
