diff -up mysql-5.5.30/mysql.init.daemonstatus mysql-5.5.30/mysql.init
--- mysql-5.5.30/mysql.init.daemonstatus	2013-04-09 15:31:21.765774369 +0200
+++ mysql-5.5.30/mysql.init	2013-04-09 15:33:37.521819540 +0200
@@ -24,7 +24,7 @@
 
 
 exec="/usr/bin/mysqld_safe"
-prog="mysqld"
+prog="__SCL_NAME__-mysqld"
 
 # Set timeouts here so they can be overridden from /etc/sysconfig/mysqld
 STARTTIMEOUT=120
@@ -61,8 +61,15 @@ mypidfile="$result"
 start(){
     [ -x $exec ] || exit 5
     # check to see if it's already running
+    MYSQLDRUNNING=0
+    if [ -f "$mypidfile" ]; then
+        MYSQLPID=`cat "$mypidfile"`
+        if [ -n "$MYSQLPID" ]; then
+            /bin/kill -0 "$MYSQLPID" && MYSQLDRUNNING=1
+        fi
+    fi
     RESPONSE=`/usr/bin/mysqladmin --socket="$socketfile" --user=UNKNOWN_MYSQL_USER ping 2>&1`
-    if [ $? = 0 ]; then
+    if [ $MYSQLDRUNNING = 1 ] && [ $? = 0 ]; then
 	# already running, do nothing
 	action $"Starting $prog: " /bin/true
 	ret=0
@@ -206,7 +213,7 @@ case "$1" in
     stop
     ;;
   status)
-    status $prog
+    status -p $mypidfile $prog
     ;;
   restart)
     restart
